Formula/kong.rb

```diff
@@ -40,40 +55,98 @@ class Kong < Formula
        end
   PATCH
 
+  def fix_dylib_references(install_map, executable_path)
+    `otool -L #{executable_path}`.scan(/(?<=\t)(.*)(?= \(.*\n)/) do |paths|
+      old_path = paths[0]
+      lib_name = old_path.sub(%r{.*/}, "")
+      new_path = install_map[lib_name]
+      # homebrew wants us to use the macho library, but it does not work here for some reason
+      system "/usr/bin/install_name_tool", "-change", old_path, new_path, executable_path if new_path
+    end
+  end
+
   def install
-    openresty_prefix = Formula["kong/kong/openresty@#{KONG_OPENRESTY_VERSION}"].prefix
+    tmpdir = format("%<prefix>s/kong-build.%<random>f.%<now>i",
+                    prefix: ENV["HOMEBREW_TEMP"],
+                    random: rand,
+                    now:    Time.now.to_i)
+    bazel_output_user_root = "--output_user_root=#{tmpdir}/bazel"
 
-    luarocks_prefix = openresty_prefix + "luarocks"
-    openssl_prefix = openresty_prefix + "openssl"
+    # Build kong, carefully setting the environment so that brew and bazel cooperate
+    python_prefix = `brew --prefix python`.strip
+    coreutils_prefix = `brew --prefix coreutils`.strip
+    path = ENV["PATH"]
 
-    bin.install_symlink "#{openresty_prefix}/openresty/nginx/sbin/nginx"
-    bin.install_symlink "#{openresty_prefix}/openresty/bin/openresty"
-    bin.install_symlink "#{openresty_prefix}/openresty/bin/resty"
-    bin.install_symlink "#{luarocks_prefix}/bin/luarocks"
+    with_env(
+      "HOME" => "#{tmpdir}/home",
+      "PATH" => "#{python_prefix}/libexec/bin:#{coreutils_prefix}/libexec/gnubin:/usr/bin:#{path}",
+    ) do
+      system "bazel",
+             bazel_output_user_root,
+             "build",
+             "--config=release",
+             "//build:kong",
+             "--action_env=HOME",
+             "--action_env=INSTALL_DESTDIR=#{prefix}",
+             "--verbose_failures"
+    end
 
-    prefix.install "kong/include"
+    prefix.install Dir["bazel-bin/build/kong-dev/*"]
+    include.install "kong/include/opentelemetry"
+
+    bin.install "bin/kong"
+    bin.install_symlink "#{prefix}/openresty/bin/opm"
+    bin.install_symlink "#{prefix}/openresty/bin/resty"
+    bin.install_symlink "#{prefix}/openresty/bin/restydoc"
+    bin.install_symlink "#{prefix}/openresty/bin/restydoc-index"
+    bin.install_symlink "#{prefix}/openresty/nginx/sbin/nginx"
+    ln_s "#{bin}/nginx", "#{bin}/openresty"
 
     yaml_libdir = Formula["libyaml"].opt_lib
     yaml_incdir = Formula["libyaml"].opt_include
 
-    system "#{luarocks_prefix}/bin/luarocks",
```

Comment:
I think the luarocks make step can be skipped now, (it's already handled in the bazel build script).